internal/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -448,21 +456,31 @@ func (c *enrollCmd) daemonReloadWithBackoff(ctx context.Context) error {
 	if err == nil {
 		return nil
 	}
+	if errors.Is(err, context.DeadlineExceeded) ||
+			errors.Is(err, context.Canceled) {
+		return fmt.Errorf("could not reload daemon: %w", err)
+	}
 
 	signal := make(chan struct{})
+	defer close(signal)
 	backExp := backoff.NewExpBackoff(signal, 10*time.Second, 1*time.Minute)
 
-	for i := 5; i >= 0; i-- {
+	for i := 0; i < 5; i++ {
```

Comment:
Where did the 5 here come from? 60s / 10s = 6 so I'd expect 6 iterations.

I think this would be simpler and clearer if we used https://github.com/cenkalti/backoff since it has native support for contexts, see example in  https://github.com/elastic/elastic-agent/blob/97e82175953af75fbe29fccc3cfcc5688a96dc8f/internal/pkg/agent/application/upgrade/step_download.go#L241-L265

Reply:
I still don't obviously see where the number 5 came from. The advantage of the package I suggested is you can just do `context.WithoutTimeout(ctx, timeout)` and limit the maximum retry time easily and also automatically handle the context errors instead of doing that separately.

We could improve our backoff implementation, but that seems like a waste when there's an off the shelf implementation that already does it (which I have used widely even prior to Elastic).

Reply:
Anyway I'm not blocking the review on changing the backoff implementation, I just think it would make this easier to follow.